apps/admin_app/lib/admin_app_web/controllers/order_controller.ex

```diff
@@ -1,26 +1,211 @@
 defmodule AdminAppWeb.OrderController do
   use AdminAppWeb, :controller
   alias Snitch.Data.Model.Order
+  alias BeepBop.Context
+  alias Snitch.Domain.Order.DefaultMachine
+  alias Snitch.Data.Schema.Variant
+  alias Snitch.Data.Schema.Address
   alias Snitch.Repo
+  import Ecto.Query
 
   def index(conn, _params) do
     render(conn, "index.html", %{orders: Repo.preload(Order.get_all(), :user)})
   end
 
-  def show(conn, %{"slug" => slug} = _params) do
-    order =
-      %{slug: slug}
-      |> Order.get()
-      |> Repo.preload(line_items: [:variant], billing_address: [], shipping_address: [])
+  def show(conn, params) do
+    order = load_order(%{number: params["number"]})
 
-    render(conn, "show.html", %{order: order})
+    search =
+      case params["search"] do
+        nil ->
+          []
+
+        _ ->
+          search_item_variant(params["search"])
+      end
+
+    render(conn, "show.html", %{order: order, search: search})
+  end
+
+  def create(conn, _params) do
+    current_user = Guardian.Plug.current_resource(conn)
+    {:ok, order} = Order.create(%{line_items: [], user_id: current_user.id})
+
+    redirect(conn, to: "/orders/#{order.number}")
   end
 
   def edit(conn, params) do
-    render(conn, "edit.html", params)
+    order = load_order(%{number: params["order_number"]})
+
+    update_item = params["update"]
+    line_items = remove_line_item(update_item, order.line_items)
+    update_item = List.first(fetch_line_item(update_item, order.line_items))
+
+    render(conn, "edit.html", %{order: order, search: [], update_item: update_item})
+  end
+
+  def update_line_item(conn, params) do
+    order = load_order(%{number: params["order_number"]})
+
+    update_item = params["update"]
+    line_items = remove_line_item(update_item, order.line_items)
+    update_item = List.first(fetch_line_item(update_item, order.line_items))
+    updated_list = [%{quantity: params["quantity"], id: update_item.id} | line_items || []]
+
+    case Order.update(%{line_items: updated_list}, order) do
+      {:ok, order} ->
+        redirect(conn, to: "/orders/#{order.number}")
+
+      {:error, error} ->
+        %{errors: [line_items: {error_text, _}]} = error
+        conn = put_flash(conn, :error, error_text)
+        redirect(conn, to: "/orders/#{order.number}")
+    end
+
+    render(conn, "edit.html", %{order: order, search: [], update_item: update_item})
+  end
+
+  def add(conn, params) do
```

Comment:
**No logic in controller please. Use context strategies**. Just in case for reference `https://github.com/aviabird/takeapledge_api/blob/master/lib/takeaplege_api/web/controllers/category_controller.ex`

Reply:
DB operation is are one at `core`, data is just parsed in this layer